ssh_check/tests/test_ssh.py

```diff
@@ -19,25 +19,39 @@
 CONNECTION_SUCCEEDED = None
 
 
-def _setup_check_with_mock_client(instance, connect_result):
+def _setup_check_with_mock_client(instance, connect_result, authentication_result=True):
+    mock_transport = MagicMock()
+    mock_transport.remote_version = 'SSH-2.0-OpenSSH_8.1'
+    mock_transport.is_authenticated.return_value = authentication_result
```

Comment:
```suggestion
def _setup_check_with_mock_client(instance, connect_result, authentication_result=None):
    mock_transport = MagicMock()
    mock_transport.remote_version = 'SSH-2.0-OpenSSH_8.1'
    mock_transport.is_authenticated.return_value = False
    if authentication_result is not None:
    	mock_transport.is_authenticated.return_value = authentication_result
```
Unless we explicity bypass authentication, by default, it should be unauthenticated.